File: Backend/BookStore/Models/Wishlists/Wishlist.cs
using System.ComponentModel.DataAnnotations;
using BookStore.Models.Users;

namespace BookStore.Models.Wishlists
{
    public class Wishlist : BaseEntity
    {
        public string AppUserId { get; set; } = string.Empty;
        public AppUser AppUser { get; set; } = null!;
        [Required]
        public string Name { get; set; } = "Default";
        public bool IsDefault { get; set; } = true;
        public string? Description { get; set; }
        public ICollection<WishlistItem> WishlistItems { get; set; } = new List<WishlistItem>();

    }
}

File: Frontend/BookStore/src/Types/User/wishlist.ts
import { BasicBookInfo } from "../book";

export interface Wishlist {
  id: number;
  name: string;
  isDefault: boolean;
  description?: string;
  wishlistItems: WishlistItem[];
}

export interface WishlistItem {
  id?: number;
  bookId: number;
  wishlistId?: number;
  book?: BasicBookInfo;
}

export interface WishlistInput {
  name: string;
  description: string;
  isDefault: boolean;
}
